@xen-orchestra/log/capture.test.js

```diff
@@ -0,0 +1,225 @@
+'use strict'
+
+const { describe, it, beforeEach } = require('node:test')
+const assert = require('node:assert/strict')
+const { createLogger } = require('@xen-orchestra/log')
+const { configure } = require('@xen-orchestra/log/configure')
+const { captureLogs } = require('@xen-orchestra/log/capture')
+const { createCaptureTransport } = require('@xen-orchestra/log/capture')
+const { dedupe } = require('@xen-orchestra/log/dedupe')
+
+describe('capture log', () => {
+  const logger = createLogger('test-logger')
+
+  const logsTransportDefault = []
+  const transportTest = message => {
+    logsTransportDefault.push(message)
+  }
+
+  const logsTransportFatal = []
+  const transportFatal = message => {
+    logsTransportFatal.push(message)
+  }
+
+  beforeEach(() => {
+    configure(
+      createCaptureTransport([
+        {
+          level: 'fatal',
+          transport: transportFatal,
+        },
+        {
+          level: 'debug',
+          transport: transportTest,
+        },
+      ])
+    )
+
+    logsTransportDefault.length = 0
+    logsTransportFatal.length = 0
+  })
+
+  it('should log test with all its attributes', () => {
+    const expected1 = {
+      level: 20,
+      namespace: 'test-logger',
+      message: 'test 1 with all its attributes',
+      time: new Date(),
+    }
+    logger.debug(expected1.message)
+    assert.equal(logsTransportDefault[0]?.message, expected1.message)
+    assert.equal(typeof logsTransportDefault[0]?.level, 'number', `level attribute should exist`)
```

Comment:
Do not bother adding an assert message unless it brings a new info.

In this case, I don't see how it would really help, if the test fails, we know which assert did and we fix it.

To be clear, it's absolutely not a problem that you provided a message, but you don't have to :slightly_smiling_face: